enumerable_methods.rb

```diff
@@ -0,0 +1,126 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    result = []
+    i = 0
+    while i < size
+      result << self[i] if yield(self[i]) == true
+      i += 1
+    end
+  end
```

Comment:
my_select...

- my_select should return an array containing all elements of enum for which the given block returns a true value.

`[1,2,3,4,5].select { |num|  num.even?  }` returns  [2, 4]
`a.select { |v| v =~ /[aeiou]/ }`  returns ["a", "e"]

Both of these tests on your code returns nil.